service/history/workflow/context.go

```diff
@@ -166,25 +174,21 @@ func (c *ContextImpl) LoadMutableState(ctx context.Context, shardContext history
 		c.MutableState = mutableState
 	}
 
-	// TODO: Use archetype ID instead of name to do the comparsion
-	// after adding archetypeID to chasm tasks as well and chasm.ArchetypeAny is removed.
-	actualArchetype, err := c.MutableState.ChasmTree().Archetype()
-	if err != nil {
-		return nil, err
-	}
-
-	if actualArchetype != "" && c.archetype != chasm.ArchetypeAny && c.archetype != actualArchetype {
+	actualArchetypeID := c.MutableState.ChasmTree().ArchetypeID()
+	if c.archetypeID != chasm.UnspecifiedArchetypeID && c.archetypeID != actualArchetypeID {
+		// TODO: convert to archetype name for better observability
 		c.logger.Warn("Potential ID conflict across different archetypes",
-			tag.Archetype(c.archetype),
-			tag.NewStringTag("actual-archetype", actualArchetype),
+			tag.ArchetypeID(c.archetypeID),
+			tag.NewUInt32("actual-archetype-id", actualArchetypeID),
```

Comment:
My concern with these logs is that they're going to be very hard to reason about. Ideally we would resolve the archetype names before logging and emitting other operational information such as metrics and internal errors.